autoload/lsp/ui/vim.vim

```diff
@@ -305,28 +306,28 @@ function! s:handle_hover(server, last_req_id, type, data) abort
 
     let l:contents = a:data['response']['result']['contents']
 
+    if empty(l:contents)
+        echom 'No ' . a:type .' found'
+        return
+    endif
+
+
     if type(l:contents) == type('')
-        let l:contents = [{ 'text': s:markdown_to_text(l:contents) }]
-    elseif type(l:contents) == type([])
-        let l:contents = []
-        for l:content in a:data['response']['result']['contents']
+        let l:contents = [l:contents]
+    endif
+
+    if type(l:contents) == type([])
+        let l:hoverText = ""
+        for l:content in l:contents
             if type(l:content) == type('')
-                call add(l:contents, { 'text': s:markdown_to_text(l:content) })
+                let l:hoverText = l:hoverText . "\n" . s:markdown_to_text(l:content)
             elseif type(l:content) == type({})
-                call add(l:contents, { 'text': s:markdown_to_text(l:content['value']) })
+                let l:hoverText = l:hoverText . "\n" . s:markdown_to_text(l:content['value'])
             endif
         endfor
-    endif
-
-    call setqflist(l:contents)
-
-    " autocmd FileType qf setlocal wrap
-
-    if empty(l:contents)
-        echom 'No ' . a:type .' found'
-    else
-        echom 'Retrieved ' . a:type
-        copen
+        let a:ctx['done'] = 1
+        " first character will always be a newline so skip it
```

Comment:
Why would the first character always be new line? Is this a bug in the lang server or a bug in the parser or part of the spec? Also would be good to have a what log file shows for the response.